iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergAcidUtil.java

```diff
@@ -196,6 +213,32 @@ public static long computePartitionHash(Record rec) {
     return computeHash(part);
   }
 
+  public static PartitionKey parsePartitionKey(Record rec) {
+    String serializedStr = rec.get(SERDE_META_COLS.get(PARTITION_PROJECTION), String.class);
+    return SerializationUtil.deserializeFromBase64(serializedStr);
+  }
+
+  public static String getSerializedPartitionKey(Record rec, PartitionSpec partitionSpec) {
+    GenericRecord current = GenericRecord.create(partitionSpec.schema());
+    for (int index = 0; index < rec.size(); index++) {
+      current.set(index, rec.get(index));
+    }
+    StructLike structLike = new InternalRecordWrapper(partitionSpec.schema().asStruct()).wrap(current);
+    PartitionKey partitionKey = new PartitionKey(partitionSpec, partitionSpec.schema());
+    partitionKey.partition(structLike);
+    return SerializationUtil.serializeToBase64(partitionKey);
+  }
+
+  public static String getSerializedPartitionKey(StructLike structLike, PartitionSpec partitionSpec) {
+    PartitionKey partitionKey = new PartitionKey(partitionSpec, partitionSpec.schema());
+    if (structLike != null) {
```

Comment:
can't we do `partitionKey.partition(structLike)`?
if yes we could reuse that method in 
````
public static String getSerializedPartitionKey(Record rec, PartitionSpec partitionSpec) {
    GenericRecord current = GenericRecord.create(partitionSpec.schema());
    for (int index = 0; index < rec.size(); index++) {
      current.set(index, rec.get(index));
    }
    StructLike structLike = new InternalRecordWrapper(partitionSpec.schema().asStruct()).wrap(current);
    return getSerializedPartitionKey(structLike, partitionSpec)
  }
````

Reply:
The `structLike` object we get contains the partition information only and not the entire row information. Hence the for-loop iteration is not correct.

Reply:
`getSerializedPartitionKey(Record rec, PartitionSpec partitionSpec)` was removed?

Reply:
Yes, that was not required at all.